pkg/blobaccess/wget/access.go

```diff
@@ -0,0 +1,128 @@
+package wget
+
+import (
+	gocontext "context"
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	"github.com/open-component-model/ocm/pkg/blobaccess"
+	"github.com/open-component-model/ocm/pkg/blobaccess/bpi"
+	"github.com/open-component-model/ocm/pkg/contexts/credentials"
+	"github.com/open-component-model/ocm/pkg/contexts/credentials/builtin/wget/identity"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi"
+	"github.com/open-component-model/ocm/pkg/errors"
+	"github.com/open-component-model/ocm/pkg/mime"
+	"github.com/open-component-model/ocm/pkg/optionutils"
+)
+
+const (
+	CACHE_CONTENT_THRESHOLD = 4096
+)
+
+func DataAccessForWget(url string, opts ...Option) (blobaccess.DataAccess, error) {
+	blobAccess, err := BlobAccessForWget(url, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return blobAccess, nil
+}
+
+func BlobAccessForWget(url string, opts ...Option) (_ blobaccess.BlobAccess, rerr error) {
+	eff := optionutils.EvalOptions(opts...)
+	log := eff.Logger("URL", url)
+
+	creds, err := eff.GetCredentials(url)
+	if err != nil {
+		return nil, err
+	}
+	if creds == nil {
+		log.Debug("no credentials found for", "url", url)
+	}
+
+	// configure http client
+	rootCAs, err := credentials.GetRootCAs(eff.CredentialContext, creds)
+	if rerr != nil {
+		return nil, err
+	}
+	clientCerts, err := credentials.GetClientCerts(eff.CredentialContext, creds)
+	if err != nil {
+		return nil, errors.New("client certificate and private key provided in credentials could not be loaded " +
+			"as tls certificate")
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			MinVersion:   tls.VersionTLS13,
+			RootCAs:      rootCAs,
+			Certificates: clientCerts,
+		},
+	}
+
+	client := &http.Client{
+		Transport: transport,
+	}
+
+	// configure http request
+	request, err := http.NewRequestWithContext(gocontext.Background(), http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if creds != nil {
+		user := creds.GetProperty(identity.ATTR_USERNAME)
+		password := creds.GetProperty(identity.ATTR_PASSWORD)
+		token := creds.GetProperty(identity.ATTR_IDENTITY_TOKEN)
+
+		if user != "" && password != "" {
+			request.SetBasicAuth(user, password)
+		} else if token != "" {
+			request.Header.Set("Authorization", "Bearer "+token)
+		}
+	}
+
+	resp, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer errors.PropagateError(&rerr, resp.Body.Close)
+	log.Debug("http status code", "", resp.StatusCode)
+
+	var blob cpi.BlobAccess
+	if resp.ContentLength < 0 || resp.ContentLength > CACHE_CONTENT_THRESHOLD {
+		log.Debug("download to file because content length is", "unknown or greater than", CACHE_CONTENT_THRESHOLD)
```

Comment:
Why is the log message splitted like this? Would it be too long?
```suggestion
		log.Debug("download to file because content length is unknown or greater than", CACHE_CONTENT_THRESHOLD)
```

Reply:
The signature of the log function is `Debug(msg string, keypairs ...interface{})`. This quite weird and I just thought I'd provide a key value pair.

Reply:
Adjusted.